### PR TITLE
[node] Add global WebSocket

### PR DESCRIPTION
**Pull request by Semigradsky**
https://nodejs.org/docs/latest-v22.x/api/globals.html#websocket

### CODE REVIEW

File: types/node/v20/globals.d.ts [hunk 1]
```diff
@@ -409,4 +410,8 @@ declare global {
         File: infer T;
     } ? T
         : typeof import("node:buffer").File;
+
+    interface WebSocket extends _WebSocket {}
+    var WebSocket: typeof globalThis extends { onmessage: any; WebSocket: infer T } ? T
+        : typeof import("undici-types").WebSocket;
```
Comment (mcollina) [inline types/node/v20/globals.d.ts:416]: This is not enabled in v20 by default.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: types/node/v20/globals.d.ts (hunk 1 above)
Comment (Semigradsky) [reply types/node/v20/globals.d.ts:416]: Yes, this is enabled via a CLI flag. What should we do in this case?

Comment (Semigradsky) [reply types/node/v20/globals.d.ts:416]: I see that we have defined `vm.Module` which only available with flag too.

https://nodejs.org/docs/latest-v20.x/api/vm.html#class-vmmodule

Comment (mcollina) [reply types/node/v20/globals.d.ts:416]: I would recommend to not type this in v20.

Comment (Semigradsky) [reply types/node/v20/globals.d.ts]: Ok, I removed types for v20

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jakebailey — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Semigradsky** [comment]
Ready to merge

**Pull request merged**